tests/test-framework/src/fetch.rs

```diff
@@ -14,27 +14,30 @@
 // You should have received a copy of the GNU General Public License
 // along with the Leo library. If not, see <https://www.gnu.org/licenses/>.
 
-use std::{fs, path::Path};
+use std::{
+    fs,
+    path::{Path, PathBuf},
+};
 
-pub fn find_tests<T: AsRef<Path>>(path: T, out: &mut Vec<(String, String)>) {
-    for entry in fs::read_dir(path).expect("fail to read tests") {
-        let entry = entry.expect("fail to read tests").path();
-        if entry.is_dir() {
-            find_tests(entry.as_path(), out);
-            continue;
-        } else if entry.extension().and_then(|x| x.to_str()).unwrap_or_default() != "leo" {
-            continue;
-        }
-        let content = fs::read_to_string(entry.as_path()).expect("failed to read test");
-        out.push((entry.as_path().to_str().unwrap_or_default().to_string(), content));
-    }
+use walkdir::WalkDir;
```

Comment:
@gluax We previously relied on `walkdir` and went through significant effort to remove the dependency.
Is it possible to reimplement without it?

Reply:
Also as a note if we wish to avoid `walkdir` currently `leo-package` directly uses it as a dep. I also see that the `notify` package has it as a sub dep.

Reply:
Yes lets remove them both.
I will open a separate PR to do so.

Reply:
Sounds good to me, seems those deps were added before I even worked here so some cleanup of those codebases is probably necessary at some point.